crates/bdk/src/wallet/mod.rs

```diff
@@ -1403,7 +1404,7 @@ impl<D> Wallet<D> {
             .index
             .index_of_spk(&txout.script_pubkey)?;
         let descriptor = self.get_descriptor_for_keychain(keychain);
-        Some(descriptor.at_derivation_index(child))
+        Some(descriptor.at_derivation_index(child).expect("TODO"))
```

Comment:
In d12242ec746ffc51da64672d7c7f7e014258d190:

These TODOs could be `.ok()?` which wil ljust return None if the child index is somehow out of range.